pandaserver/dataservice/adder_gen.py

```diff
@@ -117,411 +117,461 @@ def get_plugin_class(self, tmp_vo, tmp_group):
-                        db_lock = self.lock_pool.get(self.job.jediTaskID)
-                        if db_lock:
-                            db_lock.acquire()
-                            self.logger.debug(f"got DB lock for jediTaskID={self.job.jediTaskID}")
-                        else:
-                            self.logger.debug(f"couldn't get DB lock for jediTaskID={self.job.jediTaskID}")
-                    self.logger.debug("updating DB")
-                    update_result = self.taskBuffer.updateJobs(
-                        [self.job],
-                        False,
-                        oldJobStatusList=[old_job_status],
-                        extraInfo=self.extra_info,
-                        async_dataset_update=True,
-                    )
-                    self.logger.debug(f"retU: {update_result}")
-                    if db_lock:
-                        self.logger.debug(f"release DB lock for jediTaskID={self.job.jediTaskID}")
-                        db_lock.release()
-                        self.lock_pool.release(self.job.jediTaskID)
-                    # failed
-                    if not update_result[0]:
-                        self.logger.error(f"failed to update DB for pandaid={self.job.PandaID}")
-                        # unlock job output report
-                        self.taskBuffer.unlockJobOutputReport(
-                            panda_id=self.job_id,
-                            attempt_nr=self.attempt_nr,
-                            pid=self.pid,
-                            lock_offset=self.lock_offset,
-                        )
-                        return
+                except Exception as e:
+                    self.logger.error(f"apply_retrial_rules 2 excepted and needs to be investigated ({e}): {traceback.format_exc()}")
 
-                    try:
-                        # updateJobs was successful and it failed a job with taskBufferErrorCode
-                        self.logger.debug("AdderGen.run will peek the job")
-                        job_tmp = self.taskBuffer.peekJobs(
-                            [self.job.PandaID],
-                            fromDefined=False,
-                            fromArchived=True,
-                            fromWaiting=False,
-                        )[0]
-                        self.logger.debug(
-                            f"status {job_tmp.jobStatus}, taskBufferErrorCode {job_tmp.taskBufferErrorCode}, taskBufferErrorDiag {job_tmp.taskBufferErrorDiag}"
-                        )
-                        if job_tmp.jobStatus == "failed" and job_tmp.taskBufferErrorCode:
-                            source = "taskBufferErrorCode"
-                            error_code = job_tmp.taskBufferErrorCode
-                            error_diag = job_tmp.taskBufferErrorDiag
-                            errors = [
+                # setup for closer
```

Comment:
check_job_status got a bit long. Maybe the "closer" code could be separated?

Reply:
I was also thinking about this, I agree